src/core/Akka.Cluster/ClusterActorRefProvider.cs

```diff
@@ -121,7 +121,7 @@ public override Deploy ParseConfig(string key, Config config)
                 && !config.HasPath("nr-of-instances"))
             {
                 var maxTotalNrOfInstances = config
-                    .WithFallback(ClusterConfigFactory.Default())
+                    .WithFallback(Default)
```

Comment:
Fixes the bug reported in #2266 - correctly pulls in the `akka.actor.deplyoment.default` section and has us search for the default `cluster.max-total-nr-of-instances` inside of there, which fixes the issue.